packages/node-runtime-worker-thread/src/rpc.ts

```diff
@@ -114,7 +128,14 @@ export function createCaller<Impl extends {}>(
 ): Caller<Impl, typeof methodNames[number]> {
   const obj = {};
   methodNames.forEach((name) => {
-    (obj as any)[name] = caller(name as string, getRPCOptions(messageBus));
+    const c = caller(name as string, getRPCOptions(messageBus));
+    (obj as any)[name] = async(...args: unknown[]) => {
+      const result = await c(...args);
+      if (typeof result === 'object' && result !== null && ERROR in result) {
```

Comment:
i think `message.type === 'error'` here would do wonders :) however if you decide to not go that way, i'd pull this condition out to a function and give it a nice name.

Reply:
Yeah, you're right, refactored in 6cd0e3b and it feels more straightforward now